### DISCUSSION TITLE
Mouse events

### DISCUSSION THREAD

**Discussion by G3UKB** [emilk/egui#2227] [Q&A]
Hi. Just getting my head around egui and I can see its possible to make a nice looking UI in this but is there a way to capture mouse enter, leave and wheel events. I need to do this on a digit which can be a label to scroll the digit up/down as part of a 9 digit number.

**Comment by n00kii** [comment]
you can use `ui.rect_contains_pointer` to figure out whether or not the mouse is inside a certain area, and you can use `ui.ctx().input.events` and search for the `Zoom` (ctrl wheel) or `Scroll` (wheel) events

**Reply by G3UKB** [reply]
Thanks. I will give that a try.

**Reply by G3UKB** [reply]
Having made some progress now I wanted to get this working but having trouble digging out the event. I can find the scroll event in egui::Event::Scroll() but it seems to expect a vector parameter. I got this far.
```
let f_100M = ui.label(RichText::new(&self.f_100M).text_style(heading2())
            .size(30.0)
            .strong());
            if ui.rect_contains_pointer(f_100M.rect) {
                let e = &ui.ctx().input().events;
                egui::Event::Scroll(())
            } 
```

**Comment by G3UKB** [comment]
This works but seems a bit long winded. Maybe a better way.
```
 if ui.rect_contains_pointer(f_100M.rect) {
                let e = &ui.ctx().input().events;
                if e.len() > 0 {
                    match &e[0] {
                        egui::Event::Scroll(v) => println!("Scroll {:?}", v[1]),
                        _ => (),
                    }
                }
            } 
```

**Reply by lbirkert** [reply]
I'd search in the entire list, as there might be multiple events contained in there and scroll doesn't have to be the first.

```rs
if ui.rect_contains_pointer(rect) {
    ui.ctx().input(|i| {
        for event in &i.events {
            if let egui::Event::Scroll(v) = event {
            
            }
        }
    });
}
```
